### ISSUE TITLE
Thirsty example broken in 0.19.0

### ISSUE THREAD

**Issue by ekalosak** [zkat/big-brain#97]
The `thirst` example is broken using:
- big brain 0.19.0
- bevy 0.13.2
- stable-x86_64-unknown-linux-gnu - rustc 1.78.0 (9b00956e5 2024-04-29)

## Minimum reproducible example
https://github.com/zkat/big-brain/blob/main/examples/thirst.rs

## Error text:
```
error[E0053]: method `build` has an incompatible type for trait
   --> examples/bigbrain.rs:108:35
    |
108 | #[derive(Clone, Component, Debug, ScorerBuilder)]
    |                                   ^^^^^^^^^^^^^ expected `Commands<'_, '_>`, found a different `Commands<'_, '_>`
    |
    = note: expected signature `fn(&Thirsty, &mut bevy_ecs::system::commands::Commands<'_, '_>, bevy_ecs::entity::Entity, bevy_ecs::entity::Entity)`
               found signature `fn(&Thirsty, &mut bevy::prelude::Commands<'_, '_>, bevy::prelude::Entity, bevy::prelude::Entity)`
    = note: this error originates in the derive macro `ScorerBuilder` (in Nightly builds, run with -Z macro-backtrace for more info)
```

## Potential change implicated
https://github.com/zkat/big-brain/pull/93#discussion_r1601960566

**Comment by stargazing-dino** [comment]
So is the error that we're expecting a `Command` defined in a different module or something?

If that's the case, we should be able to change [this line](https://github.com/zkat/big-brain/blob/98337faacd5eb720cc820628fd1bfa07e6c8effa/derive/src/scorer.rs#L58) to use the bevy_ecs import one and that'd resolve it no?

**Comment by ekalosak** [comment]
It's weird, right?

Somehow the `bevy::prelude::Command` is not the `bevy::ecs::system::Command`.

This project's derive (https://github.com/zkat/big-brain/blob/main/derive/src/action.rs#L58) uses that latter. The code I used to replicate (clean project, those two dependencies, verbatim copy of thirsty.rs) uses the `bevy::prelude::*` which is the former.

Also, worth noting that cloning the project and checking the example (ostensibly with dev-dependencies) does not cause this issue, and the example works.

**Comment by ekalosak** [comment]
e.g. this, @stargazing-dino  https://github.com/zkat/big-brain/pull/99 ?

**Comment by stargazing-dino** [comment]
> Also, worth noting that cloning the project and checking the example (ostensibly with dev-dependencies) does not cause this issue, and the example works in that context.

Super weird.

Yeah, just ran it and didn't get any errors myself. also upgraded to latest stable rust and still nothing.

You're PR looks correct though. Left an approval. You'll need @zkat to run CI to verify tho

**Comment by ekalosak** [comment]
I should refine my instructions for minimal replication, sorry for the ambiguity:
```sh
cargo new foo && cd foo
cargo add bevy big-brain
# copy the thirst example to examples/bigbrain.rs
cargo check —examples bigbrain
```

Crucially, the example needs to be built using the big brain crate as an external dependency to cause the error. I’m guessing it has something to do with the difference between big brains dependencies and its dev-dependencies, but that’s only a hunch. I’m still not sure what the root cause is and would appreciate any insight, if the proposed fix is not sufficient.

**Comment by stargazing-dino** [comment]
Yup, I reporduced locally. #99 fixes it for me ^^

**Comment by ekalosak** [comment]
Could be more of a legitimate RCA: https://users.rust-lang.org/t/are-re-exported-types-distinct-from-the-original-types/63911/2

Installing Bevy 0.13 could introduce the “interesting” error where the compiler claims bevy::ecs::system::Command and bevy::prelude::Command are different upon encountering the Bevy 0.12 one that big-brain 0.19 uses.

The change in #99 appears to fix. But maybe just releasing 0.20 with Bevy 0.13 would as well..?

**Comment by ekalosak** [comment]
More on this issue: https://github.com/rust-lang/rfcs/blob/master/text/1977-public-private-dependencies.md

Continuing from above: or, duh, because the thing that fixes in #99 is the bump to Bevy 0.13 not the silly change to the import path for ecs::Command

**Comment by zkat** [comment]
that's... all really weird.

**Comment by ekalosak** [comment]
Yeah but it's weird because it's my rambling, I think - the root cause is that latest (0.19) Big Brain is using Bevy 0.12 and user code is using Bevy 0.13. Thus the compiler finds "different Commands<'_, '_>"

It just doesn't specify _how_ they're different - the import path is a red herring - they are semantically versioned different.

Will be trivially fixed with releasing Big Brain 0.20 - would be great to have a better compiler message but at least TIL.

**Comment by zkat** [comment]
0.20.0 is released
